### PR TITLE
handle none upload research

### PR DESCRIPTION
**Pull request by moonso**
This PR changes the behaviour of the CLI so that it does not exit fail if there are no research variants when uploading reseach variants.

**Expected outcome**:
The command `scout upload research` will not exit fail if there are no variants

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by moonso** [comment]
Screen Shot 2021-02-23 at 14 30 46

Deployed on hasta

File: scout/commands/load/research.py [hunk 1]
```diff
@@ -50,55 +66,45 @@ def research(case_id, institute, force):
...
-                adapter.load_variants(
-                    case_obj=case_obj,
-                    variant_type="research",
-                    category="snv",
-                    rank_threshold=default_threshold,
-                )
+        if not (force or case_obj["research_requested"]):
+            LOG.warning("research not requested, use '--force'")
+            continue
+        # Test to upload research snvs
```
Comment (dnil) [inline scout/commands/load/research.py:72]: You now have neat docstrings for the single called function - time to remove the comments!

File: scout/commands/load/research.py [hunk 2]
```diff
@@ -5,11 +5,27 @@
 import click
 from flask.cli import with_appcontext
 
+from scout.adapter import MongoAdapter
 from scout.server.extensions import store
 
 LOG = logging.getLogger(__name__)
 
 
+def upload_research_variants(
```
Comment (dnil) [inline scout/commands/load/research.py:14]: 👍

File: scout/commands/load/research.py [hunk 3]
```diff
@@ -50,55 +66,45 @@ def research(case_id, institute, force):
...
-                LOG.warning("No research files found for case %s", case_id)
-                raise click.Abort()
-            case_obj["is_research"] = True
-            case_obj["research_requested"] = False
-            adapter.update_case(case_obj, keep_date=True)
-        else:
-            LOG.warning("research not requested, use '--force'")
+        if not files:
+            LOG.warning("No research files found for case %s", case_id)
+            continue
```
Comment (dnil) [inline scout/commands/load/research.py:107]: 👍

**Review by dnil — APPROVED** [review]

File: scout/commands/load/research.py (hunk 2 above)
Comment (moonso) [reply scout/commands/load/research.py:14]: :)

File: scout/commands/load/research.py (hunk 3 above)
Comment (moonso) [reply scout/commands/load/research.py:107]: :)

File: scout/commands/load/research.py (hunk 1 above)
Comment (moonso) [reply scout/commands/load/research.py]: Fixed that. I refactored another small thing (could not keep my fingers away from my old sorrows) so you might want to have a quick look again.

File: scout/commands/load/research.py [hunk 4]
```diff
@@ -1,15 +1,51 @@
...
+        case_obj=case_obj,
+        variant_type=variant_type,
+        category=category,
+        rank_threshold=rank_treshold,
+    )
+
+
+def get_case_and_institute(
+    adapter: MongoAdapter, case_id: str, institute: Optional[str]
+) -> (str, str):
```
Comment (dnil) [inline scout/commands/load/research.py:32]: This now is in want of a doctoring, the content of which is already suggested by the comments below.

File: scout/commands/load/research.py [hunk 5]
```diff
@@ -28,16 +64,9 @@ def research(case_id, institute, force):
...
-            # There was an old way to create case ids so we need a special case to handle this
-            # Assume institute-case combo
-            splitted_case = case_id.split("-")
-            # Check if first part is institute, then we know it is the old format
-            if len(splitted_case) > 1:
-                institute_obj = adapter.institute(splitted_case[0])
-                if institute_obj:
-                    institute = institute_obj["_id"]
-                    case_id = splitted_case[1]
+        case_id, institute_id = get_case_and_institute(
```
Comment (dnil) [inline scout/commands/load/research.py:67]: 👍 Much cleaner!

**Review by dnil — APPROVED** [review]

File: scout/commands/load/research.py (hunk 4 above)
Comment (moonso) [reply scout/commands/load/research.py:32]: fixed!

**Pull request merged**